src/v/cluster/archival/segment_reupload.cc

```diff
@@ -332,8 +332,8 @@ void segment_collector::do_collect(segment_collector_mode mode) {
     }
 
     if (is_reupload_mode(mode)) {
-        align_end_offset_to_manifest(
-          _target_end_inclusive.value_or(last_collected));
+        _end_inclusive = _target_end_inclusive.value_or(last_collected);
+        align_end_offset_to_manifest();
```

Comment:
i'm having a hard time following this commit as just a refactor without semantic changes. it's probably true, but i don't see it with the changes to _end_inclusive etc.. maybe it needs a slightly longer explanation or there is more commit splitting here needed?